### PR TITLE
Add ThemeProvider for light and dark theme

### LINKED ISSUE #39: Add ThemeProvider for dark theme
# Related to user story

#2

# Notes and resources

https://styled-components.com/docs/advanced#theming

### PR DESCRIPTION
**Pull request by ncbecker**

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -17,25 +20,33 @@ const AppWrapper = styled.div`
...
-              <VotingPage />
-            </Route>
-            <ProtectedRoute path="/user" component={UserPage} />
-            <ProtectedRoute path="/login" component={MenuLogInPage} />
-            <Route path="/details/:id">
-              <EpisodeDetailsPage />
-            </Route>
-          </Switch>
-        </AppWrapper>
+        <ThemeProvider theme={theme}>
```
Comment (AlexisRoe) [inline client/src/App.js]:
```suggestion
        <ThemeProvider theme={isDarkTheme ? dark : light}>
```

File: client/src/App.js [hunk 2]
```diff
@@ -17,25 +20,33 @@ const AppWrapper = styled.div`
 `;
 
 function App() {
+  const [theme, setTheme] = useState(light);
+
+  const handleChangeTheme = () => {
+    setTheme(theme === dark ? light : dark);
+  };
+
```
Comment (AlexisRoe) [inline client/src/App.js]: i saw that in an article, mabye that helps you ...

https://jacobdcastro.com/tutorials/how-to-use-themes-with-styled-components

```suggestion
const [isDarkTheme, toggleTheme] = useState(true);

  const action = () => {
    toggleTheme((isDarkTheme = !isDarkTheme));
  };
```

**Review by AlexisRoe — COMMENTED** [review]
thumbs up, mabye that helps you along

**Comment by AlexisRoe** [comment]
make sense 🤦‍♂️

**Comment by dnnsjrng** [comment]
Choose: 'Login' or 'Skip'. Choose 'Skip' and you have to login. Does this make sense?

Screen Shot 2021-01-08 at 10 32 39

**Comment by ncbecker** [comment]
> Choose: 'Login' or 'Skip'. Choose 'Skip' and you have to login. Does this make sense?
> 
> Screen Shot 2021-01-08 at 10 32 39

Good point, I should rewrite the info bubble... it's supposed to tell the user, that it's better to log in with Spotify, since you can then save your favorites longterm. However, at the moment the app only supports the authenticated version anyway. The "skip" button doesn't work yet. I might remove it completely until the feature is supported.

**Comment by ncbecker** [comment]
Please note, that the deployment only works on the first page. Due to OAuth, the complete deployment only works in the main branch after merge.

**Comment by dnnsjrng** [comment]
DarkMode: `rgb(3, 4, 94)` as a background is very glaring on a new iphone. Maybe you could slightly desaturate the color to be more in line with dark mode environment.

**Comment by dnnsjrng** [comment]
Looks significatly better!

**Review by mhstrkmp — APPROVED** [review]
Nice and clean code. Great job 👏

**Pull request merged**